### PR TITLE
dibs: implement simple-linked-list

### PR DESCRIPTION
**Pull request by laurauzcategui**
Opened also: #384

### CODE REVIEW

**Comment by laurauzcategui** [comment]
@kytrinyx could you please review? I'm new to exercism and I was trying to figure it out if this was the right way to get started.

**Comment by behrtam** [comment]
`linked-list` is a quite similar exercise #212 which might help you while working on this new exercise. There you can see how to add the exercise to the `config.json`.
For possible test cases you could have a look at other tracks like [C#](https://github.com/exercism/xcsharp/blob/master/exercises/simple-linked-list/SimpleLinkedListTest.cs) or [ruby](https://github.com/exercism/xruby/blob/master/exercises/simple-linked-list/simple_linked_list_test.rb). We use `flake8` as linter ([details](https://github.com/exercism/xpython#code-style)). You might need to run that, to fix some of your code style problems.

**Comment by laurauzcategui** [comment]
Thanks @behrtam  👍  Fixed the issues with flake8. Will be adding test cases and other helpers with progress of the days. 

Cheers,

**Comment by kytrinyx** [comment]
@behrtam will you have time to review (and potentially merge) before the end of Hacktoberfest (Monday, end of day)?

**Comment by kytrinyx** [comment]
@laurauzcategui there's a conflict in the config.json—would you mind fixing it?

**Comment by behrtam** [comment]
> Pull requests do not have to be merged and accepted; as long as they've been opened between the very start of October 1 and the very end of October 31, they count towards a free T-shirt. – [source](https://www.digitalocean.com/company/blog/ready-set-hacktoberfest/)

So, @kytrinyx there is no rush to merge this today and I will review it probably this week.

@laurauzcategui The conflict with the `config.json` can be resolved with a rebase of `upstream/master`.

**Comment by laurauzcategui** [comment]
Hi @behrtam, @kytrinx

 Thanks. Simple-linked-list is not ready yet, but binary-search-tree I believe it is :-) 

 And as @behrtam said, there is no rush for Hacktoberfest :) I already  received the email from digitalOcean that my PRs are valid :)

**Comment by laurauzcategui** [comment]
Hi @behrtam , @kytrinyx , now it's good to go for a review. 

Happy :jack_o_lantern:  

Cheers,

**Comment by behrtam** [comment]
Beside some details my first impression is good but we do have a more abstract problem. Normally singly linked lists insert and delete at the beginning of the list therefor the complexity is O(1). This is also there behaviour mentioned in the problem readme `"push-down stacks (also called a LIFO stack; Last In, First Out)"`.
Sadly we don't have a `canonical-data.json` for this problem, that would make it more clear. As a result while the *ruby track* tests exactly for this ([test_items_are_stacked](https://github.com/exercism/xruby/blob/master/exercises/simple-linked-list/simple_linked_list_test.rb#L81-L89)), the *C# track* does it the other way around ([Two_item_list_second_value](https://github.com/exercism/xcsharp/blob/master/exercises/simple-linked-list/SimpleLinkedListTest.cs#L29-L35)).

Until we have canonical test data and we found a way to harmonize all the tracks, I would go with the readme and change your behaviour. `push` should add to the head, `pop` should take from the head, `peek` should give you the value of the head.

**Comment by laurauzcategui** [comment]
Hi @behrtam  updated based on your recommendations. Looking for a new review :) 

Cheers,

**Comment by behrtam** [comment]
Great! I should find some time to review that in the next days.

**Comment by kytrinyx** [comment]
@behrtam Did this one fall between the cracks? Would you have time to give it a look?

**Comment by behrtam** [comment]
Oh, I'm sorry. Should finde some time on Sunday.

**Comment by kytrinyx** [comment]
Awesome, thanks! 🌻

File: exercises/simple-linked-list/example.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        else:
+            new_e.next = self.head
+            self.head = new_e
+        self.size += 1
+
+    def pop(self):
+        if self.empty():
+            return None
+        else:
+            e = self.head
```
Comment (behrtam) [inline exercises/simple-linked-list/example.py]: No need for undescriptive short variable names, there is enough space for `element = self...`.

**Comment by kytrinyx** [comment]
> We have decided to use stub files for all exercises (#415) so you would need to add this here as well.

@behrtam I'd recommend documenting this in the README.

**Comment by eliaahadi** [comment]
Has this been implemented or just closed due to inactivity?

**Comment by kytrinyx** [comment]
This is closed for inactivity. It doesn't look like anyone else has implemented in the mean time (it would show up in this list on master: https://github.com/exercism/python/tree/master/exercises)

@behrtam reviewed it and asked for some changes. If you want to carry the changes and implement the fixes that @behrtam suggested that would be awesome.

Check out Jess Frazelle's blog post, The Art of Closing, especially point 4 "The Carry" if you're interested in helping out with this. https://blog.jessfraz.com/post/the-art-of-closing/

**Comment by behrtam** [comment]
**Step 1)** Fork this repository and clone it locally. If you don't know how to do this take a look at this guide: https://help.github.com/articles/fork-a-repo/

**Step 2)** Add the upstream repo as remote. Details on how to do this are here: https://help.github.com/articles/configuring-a-remote-for-a-fork/

**Step 3)** Fetch the PR into a local branch, to continue the work and later create a new PR. The steps on how to checkout a PR locally can be found here: https://help.github.com/articles/checking-out-pull-requests-locally/
Keep in mind that in this case you don't want to fetch from `origin` but from `upstream`.

So, for me this looks like this:
```
➜  python git:(master) git remote -v
origin	git@github.com:behrtam/python.git (fetch)
origin	git@github.com:behrtam/python.git (push)
upstream	https://github.com/exercism/python.git (fetch)
upstream	https://github.com/exercism/python.git (push)

➜  python git:(master) git fetch upstream pull/385/head:carry-on-pr-385
remote: Counting objects: 80, done.
remote: Total 80 (delta 21), reused 21 (delta 21), pack-reused 59
Unpacking objects: 100% (80/80), done.
From https://github.com/exercism/python
 * [new ref]         refs/pull/385/head -> carry-on-pr-385

➜  python git:(master) git checkout carry-on-pr-385
Switched to branch 'carry-on-pr-385'

➜  python git:(carry-on-pr-385) python3 test/check-exercises.py simple-linked-list
.....
----------------------------------------------------------------------
Ran 5 tests in 0.000s

OK

➜  python git:(carry-on-pr-385)
```

**Comment by eliaahadi** [comment]
Thanks for information, I did steps 1 and 2. However, the checkout didn't work. The images below said local changes would be overwritten by checkout.

image

image

Any thoughts?

**Pull request closed without merging**